src/_pytest/config/__init__.py

```diff
@@ -1650,6 +1651,66 @@ def getvalueorskip(self, name: str, path=None):
         """Deprecated, use getoption(skip=True) instead."""
         return self.getoption(name, skip=True)
 
+    #: Verbosity for failed assertions (see :confval:`verbosity_assertions`).
+    VERBOSITY_ASSERTIONS: Final = "assertions"
+    _KNOWN_VERBOSITY_TYPES: Final = {VERBOSITY_ASSERTIONS}
+    _VERBOSITY_INI_DEFAULT = "auto"
+
+    def get_verbosity(self, verbosity_type: Optional[str] = None) -> int:
+        r"""Access to fine-grained verbosity levels.
+
+        .. code-block:: ini
+
+            # content of pytest.ini
+            [pytest]
+            verbosity_assertions = 2
+
+        .. code-block:: console
+
+            pytest -v
+
+        .. code-block:: python
+
+            print(config.get_verbosity())  # 1
+            print(config.get_verbosity(Config.VERBOSITY_ASSERTIONS))  # 2
+        """
+        global_level = self.option.verbose
+        assert isinstance(global_level, int)
+        if (
+            verbosity_type is None
+            or verbosity_type not in Config._KNOWN_VERBOSITY_TYPES
+        ):
+            return global_level
+
+        level = self.getini(Config._ini_name(verbosity_type))
+
+        if level == Config._VERBOSITY_INI_DEFAULT:
+            return global_level
+
+        return int(level)
+
+    @staticmethod
+    def _ini_name(verbosity_type: str) -> str:
+        return f"verbosity_{verbosity_type}"
+
+    @staticmethod
+    def _add_ini(parser: "Parser", verbosity_type: str, help: str) -> None:
```

Comment:
Same here, the name `_add_ini` is too general now.

Also, IMO this function doesn't add much value, I think it would be better to reduce the indirection and inline it. But since it's private then it's OK if you want to keep it.

Reply:
I prefer to keep it because it will avoid duplication as more types are added. It also ensures that the same default is used across verbosity types. If someone adds an Nth type, but uses "global", then `get_verbosity()` will raise an exception.